### DISCUSSION TITLE
TZE204_9mjy74mp

### DISCUSSION THREAD

**Discussion by Pejot26** [Koenkk/zigbee2mqtt#24821] [Devices]
Hello! I am a new user of HA and zigbee2mqtt. I have searched various sources, but unfortunately I havn't been able to find a solution to my problem. It is about a thermostatic head of the Moes brand with the identifier "_TZE204_9mjy74mp". I have used all methods, but these devices still have no support! Please help and be understanding for a layman.

**Comment by PuckleGit** [comment]
Hello! Quick one on the thermostatic head, do you have the specific product name MOES has given it and is it on the supported list of devices?

What you're seeing is a very common identifier for Tuya devices (lots of shared hardware with different brand names on the front). Worst case scenario you will need to build a converter to get it to work but there's a good set of instructions located [here](https://www.zigbee2mqtt.io/advanced/support-new-devices/02_support_new_tuya_devices.html)

**Comment by Pejot26** [comment]
Hello! This product is named with Moes as WTRV-PJ-801-DB28. I try add support from https://github.com/DonaldChung-HK/z2m-moes-trv-801-converter/blob/main/moes_valve_ZTRV801.js, but not works. On Ali:
https://a.aliexpress.com/_EwLGvK5.
I have 10 pieces - 7 of them have a different identifier and are supported, unfortunately the other 3 have this designation and are not supported. I've been trying for almost a month now - to no avail. Please help.

**Comment by Pejot26** [comment]
![IMG_20241119_155126](https://github.com/user-attachments/assets/9923c92b-d8b5-4f8a-9a85-7adc8f5bbd48)

As you can see, nothing helps😮‍💨

**Comment by PuckleGit** [comment]
On your list of Z2M devices, does it say if numbers 43. 44 and 45 say 'Supported' after you add the converter to your Z2M implementation?

I assume you've changed the manufactuerName on line 19 from _TZE200 to _TZE204 as per what your device is saying?

**Reply by Pejot26** [reply]
Yes I do this - rename in line 19. But nothing. The devices still are not supported. Zigbee2mqqt loads the file - this usually ends up with the add-on crashing, or after restarting the file is empty or missing as an extension.

**Reply by PuckleGit** [reply]
Hmm, that's some odd behaviour. I do wonder if some of the data-points are slightly different between the TZE200 and TZE204. A bit weird that you seeing both Z2M crashing and/or your converter.js being emptied.

I've been trying to build a thermostat for Z2M that uses the TZE200 as well and my Z2M has never crashed or deleted the data in the converter. worst case I've seen is the device going to Unsupported but usually due to bad syntax in my converter

**Reply by Pejot26** [reply]
I'm even thinking about buying new heads, hoping that I won't receive them with the same ID, because I've tried everything - every converter for these heads causes the same errors.
